tests/filtered_relation/tests.py

```diff
@@ -685,6 +703,93 @@ def test_eq(self):
+                ),
+            ).filter(
+                the_book__isnull=False,
+            ),
+            [],
+        )
+
+    def test_expression_outside_relation_name(self):
+        msg = (
+            "FilteredRelation's condition doesn't support relations outside "
+            "the 'book__editor' (got 'book__author__name')."
+        )
+        with self.assertRaisesMessage(ValueError, msg):
+            Author.objects.annotate(
+                book_editor=FilteredRelation(
+                    "book__editor",
+                    condition=Q(
+                        Exact(F("book__author__name"), "Alice"),
+                        Value(True),
+                        book__title__startswith="Poem",
+                    ),
+                ),
+            )
+
+    def test_conditional_expression_with_case(self):
+        msg = (
+            "FilteredRelation's condition doesn't support relations outside "
+            "the 'book__editor' (got 'book__author__name')."
+        )
+        with self.assertRaisesMessage(ValueError, msg):
+            Author.objects.annotate(
+                book_editor=FilteredRelation(
+                    "book__editor",
+                    condition=Q(
+                        Case(
+                            When(
+                                book__author__name="Alice",
+                                then=True,
+                            ),
+                            default=False,
+                        )
+                    ),
+                ),
+            )
+
+    def test_conditional_expression_with_func(self):
+        msg = (
+            "FilteredRelation's condition doesn't support relations outside "
+            "the 'book__editor' (got 'book__author__name')."
+        )
+        with self.assertRaisesMessage(ValueError, msg):
+            Author.objects.annotate(
+                book_editor=FilteredRelation(
+                    "book__editor",
+                    condition=Q(
+                        book__title=Func(F("book__author__name"), function="LOWER")
+                    ),
+                ),
+            )
+
+    def test_conditional_expression_with_expressionwrapper(self):
+        msg = (
+            "FilteredRelation's condition doesn't support relations outside "
+            "the 'book__editor' (got 'book__author__name')."
+        )
+        with self.assertRaisesMessage(ValueError, msg):
+            Author.objects.annotate(
+                book_editor=FilteredRelation(
+                    "book__editor",
+                    condition=Q(
+                        ExpressionWrapper(
+                            Q(
+                                Value(True),
+                                Exact(F("book__author__name"), "Alice"),
+                            ),
+                            output_field=BooleanField(),
+                        )
+                    ),
+                ),
+            )
```

Comment:
We added new tests for `ValueError`, can we also add tests with lookups, `Case()`, `ExpressionWrapper` that don't raise `ValueError`?